node/src/models/session-data.ts

```diff
@@ -0,0 +1,130 @@
+import sha256 from 'crypto-js/sha256';
+import 'models/opentutor-response';
+import { v4 as uuidv4 } from 'uuid';
+import AutoTutorData from './autotutor-data';
+import { ClassifierResult } from './classifier';
+import OpenTutorResponse, { TextData } from 'models/opentutor-response';
+
+const SESSION_SECURITY_KEY =
+  process.env.SESSION_SECURITY_KEY || 'qLUMYtBWTVtn3vVGtGZ5';
+
+export interface SessionData {
+  sessionId: string;
+  sessionHistory: SessionHistory;
+  previousUserResponse: string;
+  previousSystemResponse: string[];
+  dialogState: DialogState;
+}
+
+export interface SessionDto {
+  sessionId: string;
+  sessionHistory: string;
+  previousUserResponse: string;
+  previousSystemResponse: string[];
+  dialogState: DialogState;
+  hash: string;
+}
+
+export interface DialogState {
+  expectationsCompleted: boolean[];
+  expectationData: ExpectationData[];
+  hints: boolean;
+}
+
+export interface ExpectationData {
+  ideal: string;
+  score: number;
+  satisfied: boolean;
+  status: string;
```

Comment:
same comment here as for `OpenTutorResponse.type`, let's make status some kind of constrained enum(ish) type instead of a plain string